src/nft/components/collection/Activity.tsx

```diff
@@ -0,0 +1,185 @@
+      <Box display={{ sm: 'none', xxl: 'block' }}>{ColumnHeaders.To}</Box>
+      <Box display={{ sm: 'none', lg: 'block' }}>Buy</Box>
+    </Box>
+  )
+}
+
+interface ActivityProps {
+  contractAddress: string
+  rarityVerified: boolean
+  collectionName: string
+}
+
+const initialFilterState = {
+  [ActivityEventType.Listing]: true,
+  [ActivityEventType.Sale]: true,
+  [ActivityEventType.Transfer]: false,
+  [ActivityEventType.CancelListing]: false,
+}
+
+const reduceFilters = (state: typeof initialFilterState, action: { eventType: ActivityEventType }) => {
+  return { ...state, [action.eventType]: !state[action.eventType] }
+}
+
+const baseHref = (event: ActivityEvent) => `/#/nfts/asset/${event.collectionAddress}/${event.tokenId}?origin=activity`
+
+export const Activity = ({ contractAddress, rarityVerified, collectionName }: ActivityProps) => {
+  const [activeFilters, filtersDispatch] = useReducer(reduceFilters, initialFilterState)
+
+  const {
+    data: eventsData,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+    isSuccess,
+    isLoading,
+  } = useInfiniteQuery<ActivityEventResponse>(
+    [
+      'collectionActivity',
+      {
+        contractAddress,
+        activeFilters,
+      },
+    ],
+    async ({ pageParam = '' }) => {
+      return await ActivityFetcher(
+        contractAddress,
+        {
+          eventTypes: Object.keys(activeFilters)
+            .filter((key) => activeFilters[key as ActivityEventType])
+            .map((key) => key as ActivityEventType),
+        },
+        pageParam
+      )
+    },
+    {
+      getNextPageParam: (lastPage) => {
+        return lastPage.events?.length === 25 ? lastPage.cursor : undefined
+      },
+      refetchInterval: 15000,
+      refetchIntervalInBackground: false,
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+    }
+  )
+
+  const events = useMemo(
+    () => (isSuccess ? eventsData?.pages.map((page) => page.events).flat() : null),
+    [isSuccess, eventsData]
+  )
+
+  const itemsInBag = useBag((state) => state.itemsInBag)
+  const addAssetToBag = useBag((state) => state.addAssetToBag)
+  const removeAssetFromBag = useBag((state) => state.removeAssetFromBag)
+  const cartExpanded = useBag((state) => state.bagExpanded)
+  const toggleCart = useBag((state) => state.toggleBag)
+  const isMobile = useIsMobile()
+  const [ethPriceInUSD, setEthPriceInUSD] = useState(0)
+
+  useEffect(() => {
+    fetchPrice().then((price) => {
+      setEthPriceInUSD(price || 0)
+    })
+  }, [])
+
+  const Filter = useCallback(
+    function ActivityFilter({ eventType }: { eventType: ActivityEventType }) {
+      const isActive = activeFilters[eventType]
+
+      return (
+        <Box
+          className={clsx(styles.filter, isActive && styles.activeFilter)}
+          onClick={() => filtersDispatch({ eventType })}
+        >
+          {eventType.charAt(0) + eventType.slice(1).toLowerCase() + 's'}
```

Comment:
nit: If the purpose of this is to capitalize then I think css is better and easier to read

```text-transform: capitalize```

Reply:
this really doesn't work that well because `captitalize` doesn't work on all words that are all caps and the `div` that is rendered is for the size of the all caps word